src/features/goblins/trader/selling/components/Drafting.tsx

```diff
@@ -82,18 +84,24 @@ export const Drafting: React.FC<DraftingProps> = ({
   };
 
   const currentInventoryAmount = inventory[selected] || new Decimal(0);
-  const maxSellAmount = itemLimits[selected];
+  const maxSellAmount = itemLimits[selected] ?? 0;
 
   const disableListTradeButton =
-    sflAmount === 0 ||
-    resourceAmount === 0 ||
+    !sflAmount ||
+    !resourceAmount ||
     new Decimal(sflAmount).gt(MAX_SFL) ||
     new Decimal(resourceAmount).gt(currentInventoryAmount);
 
+  const hasItemsToList = !!inventoryItems && inventoryItems.length > 0;
```

Comment:
Any specific reason we have added !!inventoryItems in the condition?

As filter will return an empty array if no element matches the criteria, and !! of an empty array will always be true.

Reply:
You are right, InventoryItems wont be null.  I removed the null check.